.github/workflows/convert.yml

```
@@ -0,0 +1,56 @@
+name: convert precommit and test
+run-name: convert precommit and test
+on:
+  pull_request:
+    paths:
+      - convert/**
+      - .github/workflows/convert.yml
+jobs:
+  convert-pre-commit-actions:
+    strategy:
+      matrix:
+        python-version: [ "3.8.15" ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "pip"
+      - name: Run isort
+        uses: isort/isort-action@v1.1.0
+        with:
+          sort-paths: convert/*
+          configuration:
+            --check-only
+            --profile=black
+            --line-length=79
+            --skip=tests
+      - name: Black
+        uses: psf/black@stable
+        with:
+          options: "--line-length=79"
+      - run: pip install flake8
+      - run: flake8 --extend-ignore=E203 convert
+  convert-build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.8.15" ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install Poetry
+        run: pip install poetry==1.3.0
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: "poetry"
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          cd convert
+          poetry install --all-extras
+          poetry add --editable ../lib/filter_lib
+          poetry add --editable ../lib/tenants
+          poetry run pytest
```

Here's an example how you could implement the comment above
```suggestion
jobs:
  convert:
    strategy:
      matrix:
        python-version: [ "3.8.15" ]
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-python@v4
        with:
          python-version: ${{ matrix.python-version }}
          cache: "poetry"
      - name: Install dependencies
         working-directory: ./convert
         run: |
           python -m pip install --upgrade pip
           poetry install --all-extras
           poetry add --editable ../lib/filter_lib
           poetry add --editable ../lib/tenants
      - name: Run linters and checkers [isort -> black -> mypy -> pylint]
         working-directory: ./convert
         run: |
          git ls-files -- . | xargs pre-commit run --files
       - name: Run tests
          working-directory: ./convert
          run: |
            poetry run pytest tests/
```

---

Thank you. 
Is it clear, why there is python 3.8.15 ?

---

Did not quite got your question

---

The use of version 3.8.10 of Python in our images resulted in this question.